pkg/mic/mic.go

```diff
@@ -120,7 +120,7 @@ func NewMICClient(cloudconfig string, config *rest.Config, isNamespaced bool, sy
 
 	var immutableUserMSIsMap map[string]bool
 
-	if immutableUserMSIsList != nil {
+	if immutableUserMSIsList != nil && len(immutableUserMSIsList) > 0 {
```

Comment:
Isn't len check enough since that checks for nil ?

Reply:
Done.